test/integration/associations/belongs-to.test.js

```diff
@@ -152,6 +152,44 @@ describe(Support.getTestDialectTeaser('BelongsTo'), () => {
         });
       });
     });
+
+    it('supports schemas when defining custom foreign key attribute #9029', function() {
+      const self = this,
+        User = this.sequelize.define('UserXYZ', {
+          uid: {
+            type: Sequelize.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+            allowNull: false
+          }
+        }).schema('archive')
+        , Task = this.sequelize.define('TaskXYZ', {
+          user_id: {
+            type: Sequelize.INTEGER,
+            references: { model: User, key: 'uid' }
+          }
+        }).schema('archive');
+
+      Task.belongsTo(User, { foreignKey: 'user_id'});
+
+      return self.sequelize.dropAllSchemas().then(() => {
+        return self.sequelize.createSchema('archive');
+      }).then(() => {
+        return User.sync({force: true });
```

Comment:
`User.sync` / `Task.sync` can be replaced with `this.sequelize.sync`

Reply:
schema is set using .schema() so .sync has to be called on each model separately according to [this](https://github.com/sequelize/sequelize/blob/master/lib/model.js#L1226).